app/controllers/concerns/starter_code.rb

```diff
@@ -1,8 +1,21 @@
 module StarterCode
+  include GitHub
+  extend ActiveSupport::Concern
+
+  # rubocop:disable MethodLength
   def starter_code_repository_id(repo_name)
     return unless repo_name.present?
-    sanitized_repo_name = repo_name.gsub(/\s+/, '')
-    github_repository   = GitHubRepository.new(current_user.github_client, nil)
-    github_repository.repository(sanitized_repo_name).id
+
+    if repo_name =~ %r{/[a-zA-Z0-9_-]+\/[a-zA-Z0-9_-]+/}
```

Comment:
I think you might need begin `^` and end `$` anchors in this regex. As it stands I think it might match anything containing `owner/name`.